tasks/main.yml

```diff
@@ -1,6 +1,23 @@
 ---
 
-- name: Updating cache
+- name: Check server uptime
+  shell: cat /proc/uptime
+  register: host_uptime
+  changed_when: false
+  tags:
+    - packages
+
+- name: Update cache if server has been up less than an hour
+  become: yes
+  apt:
+    update_cache: yes
+    # Make idempotency test happy
+    cache_valid_time: 60
+  when: "(host_uptime.stdout_lines[0].split(' ')[0] | int ) / 3600 < 1"
```

Comment:
I would like to see this comparison done in seconds against an appropriately named variable. The variable should have a value of `3600` (one hour) set as a default inside the packages role.